File: daily/Binary Tree/94-Binary-tree-inorder-traversal.go
package daily

func inorderTraversal(root *TreeNode) []int {
	var results []int
	dfsInorder(root, &results)
	return results
}

func dfsInorder(node *TreeNode, results *[]int) {
	if node == nil {
		return
	}
	dfsInorder(node.Left, results)
	*results = append(*results, node.Val)
	dfsInorder(node.Right, results)
}

File: daily/Binary Tree/94-Binary-tree-inorder-traversal.py
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        results = []
        def dfs(node):
            if not node:
                return
            dfs(node.left)
            results.append(node.val)
            dfs(node.right)
        dfs(root)
        return results